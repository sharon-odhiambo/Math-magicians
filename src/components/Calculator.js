import React, { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [state, setInput] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (e) => {
    e.preventDefault();
    const buttonName = e.target.value;
    const { total, next, operation } = calculate(state, buttonName);
    setInput({ total, next, operation });
  };

  const { total, next, operation } = state;

  return (
    <div className="calc">
      <div>Let&apos;s do some Math</div>
      <form>
        <div className="result">{!total && !next && !operation ? 0 : [total, operation, next]}</div>
        <table>
          <tr>
            <td><input type="button" onClick={handleClick} value="AC" /></td>
            <td><input type="button" onClick={handleClick} value="+/-" /></td>
            <td><input type="button" onClick={handleClick} value="%" /></td>
            <td><input type="button" onClick={handleClick} className="result-end" value="&#247;" /></td>
          </tr>
          <tr>
            <td><input type="button" onClick={handleClick} value="7" /></td>
            <td><input type="button" onClick={handleClick} value="8" /></td>
            <td><input type="button" onClick={handleClick} value="9" /></td>
            <td><input type="button" onClick={handleClick} className="result-end" value="x" /></td>
          </tr>
          <tr>
            <td><input type="button" onClick={handleClick} value="4" /></td>
            <td><input type="button" onClick={handleClick} value="5" /></td>
            <td><input type="button" onClick={handleClick} value="6" /></td>
            <td><input type="button" onClick={handleClick} className="result-end" value="-" /></td>
          </tr>
          <tr>
            <td><input type="button" onClick={handleClick} value="1" /></td>
            <td><input type="button" onClick={handleClick} value="2" /></td>
            <td><input type="button" onClick={handleClick} value="3" /></td>
            <td><input type="button" onClick={handleClick} className="result-end" value="+" /></td>
          </tr>
          <tr>
            <td><input type="button" onClick={handleClick} value="0" /></td>
            <td><input type="button" onClick={handleClick} value="." /></td>
            <td><input type="button" onClick={handleClick} className="result-end" value="=" /></td>
          </tr>
        </table>
      </form>
    </div>
  );
};

export default Calculator;
