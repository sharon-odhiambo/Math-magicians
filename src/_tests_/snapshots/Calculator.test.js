import React from 'react';
import { render } from '@testing-library/react';
import Calculator from '../../components/Calculator';

describe('Calculator', () => {
  test('should match Calculator snapshot', () => {
    const tree = render(
      <Calculator />,
    );
    expect(tree).toMatchSnapshot();
  });
});
