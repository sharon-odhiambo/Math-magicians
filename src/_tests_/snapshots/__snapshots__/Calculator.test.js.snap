// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator should match Calculator snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="calc"
      >
        <div>
          Let's do some Math
        </div>
        <form>
          <div
            class="result"
          >
            0
          </div>
          <table>
            <tbody>
              <tr>
                <td>
                  <input
                    type="button"
                    value="AC"
                  />
                </td>
                <td>
                  <input
                    type="button"
                    value="+/-"
                  />
                </td>
                <td>
                  <input
                    type="button"
                    value="%"
                  />
                </td>
                <td>
                  <input
                    class="result-end"
                    type="button"
                    value="รท"
                  />
                </td>
              </tr>
              <tr>
                <td>
                  <input
                    type="button"
                    value="7"
                  />
                </td>
                <td>
                  <input
                    type="button"
                    value="8"
                  />
                </td>
                <td>
                  <input
                    type="button"
                    value="9"
                  />
                </td>
                <td>
                  <input
                    class="result-end"
                    type="button"
                    value="x"
                  />
                </td>
              </tr>
              <tr>
                <td>
                  <input
                    type="button"
                    value="4"
                  />
                </td>
                <td>
                  <input
                    type="button"
                    value="5"
                  />
                </td>
                <td>
                  <input
                    type="button"
                    value="6"
                  />
                </td>
                <td>
                  <input
                    class="result-end"
                    type="button"
                    value="-"
                  />
                </td>
              </tr>
              <tr>
                <td>
                  <input
                    type="button"
                    value="1"
                  />
                </td>
                <td>
                  <input
                    type="button"
                    value="2"
                  />
                </td>
                <td>
                  <input
                    type="button"
                    value="3"
                  />
                </td>
                <td>
                  <input
                    class="result-end"
                    type="button"
                    value="+"
                  />
                </td>
              </tr>
              <tr>
                <td>
                  <input
                    type="button"
                    value="0"
                  />
                </td>
                <td>
                  <input
                    type="button"
                    value="."
                  />
                </td>
                <td>
                  <input
                    class="result-end"
                    type="button"
                    value="="
                  />
                </td>
              </tr>
            </tbody>
          </table>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="calc"
    >
      <div>
        Let's do some Math
      </div>
      <form>
        <div
          class="result"
        >
          0
        </div>
        <table>
          <tbody>
            <tr>
              <td>
                <input
                  type="button"
                  value="AC"
                />
              </td>
              <td>
                <input
                  type="button"
                  value="+/-"
                />
              </td>
              <td>
                <input
                  type="button"
                  value="%"
                />
              </td>
              <td>
                <input
                  class="result-end"
                  type="button"
                  value="รท"
                />
              </td>
            </tr>
            <tr>
              <td>
                <input
                  type="button"
                  value="7"
                />
              </td>
              <td>
                <input
                  type="button"
                  value="8"
                />
              </td>
              <td>
                <input
                  type="button"
                  value="9"
                />
              </td>
              <td>
                <input
                  class="result-end"
                  type="button"
                  value="x"
                />
              </td>
            </tr>
            <tr>
              <td>
                <input
                  type="button"
                  value="4"
                />
              </td>
              <td>
                <input
                  type="button"
                  value="5"
                />
              </td>
              <td>
                <input
                  type="button"
                  value="6"
                />
              </td>
              <td>
                <input
                  class="result-end"
                  type="button"
                  value="-"
                />
              </td>
            </tr>
            <tr>
              <td>
                <input
                  type="button"
                  value="1"
                />
              </td>
              <td>
                <input
                  type="button"
                  value="2"
                />
              </td>
              <td>
                <input
                  type="button"
                  value="3"
                />
              </td>
              <td>
                <input
                  class="result-end"
                  type="button"
                  value="+"
                />
              </td>
            </tr>
            <tr>
              <td>
                <input
                  type="button"
                  value="0"
                />
              </td>
              <td>
                <input
                  type="button"
                  value="."
                />
              </td>
              <td>
                <input
                  class="result-end"
                  type="button"
                  value="="
                />
              </td>
            </tr>
          </tbody>
        </table>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
